Make August 2024 News Public (#28479)
idaholab/moose: modules/doc/content/newsletter/2024/2024_08.md

# MOOSE Newsletter (August 2024)

## MOOSE Improvements

> The [SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,

Author: ```suggestion
### Warnings and warning tallies now available for SolutionInvalidInterface

[SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,
```

similar to the `solutionInvalid`-cases, but do not cause the solution to be rejected. Knowing the number of instances

> that a warning was triggered on is helpful for the modeler to decide manually whether the solution is acceptable. 

Author: ```suggestion
that a warning was triggered is helpful for the modeler to decide manually whether the solution is acceptable. 
```

# MOOSE Newsletter (August 2024)

## OpenMPI added to Conda MOOSE Packages alongside MPICH

> You may now specify either OpenMPI or the MPICH MPI wrapper when installing MOOSE Conda packages by appending `openmpi` or `mpich` to the `conda create/install` line:

Author: ```suggestion
You may now specify either OpenMPI or the MPICH MPI wrapper when installing MOOSE Conda packages by appending `openmpi` or `mpich` to the `conda create/install` line:

```

```bash
conda create -n moose moose-dev openmpi||mpich
```

> *Note: OpenMPI being a new package addition, is a bit under-tested at the moment.*

Author: ```suggestion

*Note: OpenMPI, being a new package addition, is a bit under-tested at the moment. It is recommended for use only by experienced users and developers!*
```

## MOOSE Improvements

### Warnings and warning tallies now available for SolutionInvalidInterface

[SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,
similar to the `solutionInvalid`-cases, but do not cause the solution to be rejected. Knowing the number of instances
that a warning was triggered is helpful for the modeler to decide manually whether the solution is acceptable.
These warnings are created using the `flagSolutionWarning` routine.

### Added fluid properties class for a two-phase fluid mixed with a non-condensable gas

The [TwoPhaseNCGPartialPressureFluidProperties.md] class was added, which derives

> from `TwoPhaseNCGFluidProperties` and takes any `TwoPhaseFluidProperties` and
> `SinglePhaseFluidProperties` as input and creates an [IdealRealGasMixtureFluidProperties.md],

Author: ```suggestion
from `TwoPhaseNCGFluidProperties`, takes any `TwoPhaseFluidProperties` and
`SinglePhaseFluidProperties` as input, and creates an [IdealRealGasMixtureFluidProperties.md],
```

# MOOSE Newsletter (August 2024)

## OpenMPI added to Conda MOOSE Packages alongside MPICH

You may now specify either OpenMPI or the MPICH MPI wrapper when installing MOOSE Conda packages by appending `openmpi` or `mpich` to the `conda create/install` line:

```bash
conda create -n moose moose-dev openmpi||mpich
```

*Note: OpenMPI, being a new package addition, is a bit under-tested at the moment. It is recommended for use only by experienced users and developers!*

## MOOSE Improvements

### New system: Correctors

The [Corrector system](syntax/Correctors/index.md) was introduced to offer first-class support for modifying
solution variables (also called nonlinear variables, except when using the linear finite volume discretization).
This can be for re-normalization purposes, for shifting the entire variable, and so on. `Correctors` derive from

> `UserObjects`, they are simply specialized for this purpose and use the `[Corrector]` syntax.

Author: ```suggestion
`UserObjects`; they are simply specialized for this purpose and use the `[Corrector]` syntax.
```

We are curious to see how users seek to modify the output of the nonlinear solver, so please let us know how you use this new system!

### New system: MeshModifiers

The [MeshModifier system](syntax/MeshModifiers/index.md) was introduced to offer first-class support for modifying
the mesh during a simulation. This can be to change the subdomain id of certain elements, to add or remove sides from a sideset, and so on.

> `MeshModifiers` derive from `UserObjects`, they are simply specialized for this purpose and use the `[MeshModifier]` syntax.

Author: ```suggestion
`MeshModifiers` derive from `UserObjects`; they are simply specialized for this purpose and use the `[MeshModifier]` syntax.
```

We are curious to see how users seek to modify the mesh during a simulation, so please let us know how you use this new system!

!alert note
`MeshModifier` used to be the old name for `MeshGenerators`. If you have a really old input file, you may see mesh modifiers with similar
names to the mesh generators. The correct fix will be to change the names of the objects in the input file to their corresponding mesh generator.

### Warnings and warning tallies now available for SolutionInvalidInterface

[SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,
similar to the `solutionInvalid`-cases, but do not cause the solution to be rejected. Knowing the number of instances
that a warning was triggered is helpful for the modeler to decide manually whether the solution is acceptable.
These warnings are created using the `flagSolutionWarning` routine.

### Block restriction debug output

> Block restriction of objects can now be output using an option in the [Debug block](SetupDebugAction.md).

Author: ```suggestion
Block restriction of objects can now be output using the [!param](/Debug/SetupDebugAction/show_block_restriction) parameter in the [Debug block](SetupDebugAction.md).
```

### MOOSE server improvements

These improvements will be available on the next update of the VSCode MOOSE plugin.

- When writing a new object in an input, the required parameters are automatically added to the input file. The user still has to fill in their value.

> - `!include` directives are no longer removed by automatic formatting
> - Inputs with a broken format will no longer be subject to automatic formatting, which would not improve the broken format
> - Parameters with a specific string type, for example `FunctionName`, will now automatically suggest the valid inputs from the rest of the input file, for example the list of `Functions` already specified in the input

Author: ```suggestion
- `!include` directives are no longer removed by automatic formatting.
- Inputs with a broken format will no longer be subject to automatic formatting, which would not improve the broken format.
- Parameters with a specific string type, for example `FunctionName`, will now automatically suggest the valid inputs from the rest of the input file. In this example, the plugin might suggest the list of `Functions` already specified in the input.
```

Consequently, the `moose-dev` conda environment was updated to version `2024.08.05`. All `moose-dev`
packages versioned after this date should contain PETSc 3.21.4. To obtain a newer version of the MOOSE
development packages, please follow our [getting started instructions](getting_started/index.md)
and create a new MOOSE conda environment. Alternatively, one may also attempt an update of their
current conda environment by performing the following commands, assuming that your development
environment is named `moose`:

```bash
conda activate moose
conda update moose-dev
```

## Bug Fixes and Minor Enhancements

- The [Terminator.md] can now output messages without failing the time step or returning an error,
  with its new 'NONE' option to the [!param](/UserObjects/Terminator/fail_mode) parameter.
- `VectorNeumannBC` was renamed to [DirectionalNeumannBC.md] for clarity.
- `MeshExtruderGenerator` was deprecated to prepare for its removal, as the [AdvancedExtruderGenerator.md]
  offers all its functionalities and more.
- `MaterialVectorPostprocessor` was renamed to [ElementMaterialSampler.md] to follow the naming convention
  used for the other vectorpostprocessor sampler objects.

> - [TransientMultiApp.md] now always restore by default on the end of fixed point iterations. Not restoring
>   was an optimization that is not valid if there are stateful properties, or generally quantities computed on
>   `TIMESTEP_BEGIN`.

Author: ```suggestion
- [TransientMultiApp.md] now always restores by default at the end of fixed point iterations. Not restoring
  was an optimization that is not valid if there are stateful properties or, generally, quantities computed on
  `TIMESTEP_BEGIN`.
```
